app/services/event_service.rb

```diff
@@ -0,0 +1,16 @@
+class EventService
+
+  def self.find_all(zipcode)
+    conn = Faraday.new(url: "http://api.eventful.com/json/events/search") do |faraday|
+      faraday.adapter Faraday.default_adapter
+    end
+    response = conn.get do |req|
+      req.params[:app_key] = "RwHdW4tXKrczPzjZ"#would be hidden in a real app
+      req.params[:location] = "#{zipcode}"
+      req.params[:within] = 10
+      req.params[:date] = "March"
+      req.params[:sort_order] = 'popularity'
+    end
+    events = JSON.parse(response.body)['events']
```

Comment:
if you were to add `['event']` at the end here and symbolize names, i believe you could have avoided needing to call `[1]` in your `Events.find_all` method and having to call `[0]` within your view. it would've looked like this:

`events = JSON.parse(response.body, symbolize_names: true)[:events][:event]`

potentially a refactor 👍